railties/lib/rails/generators/rails/app/templates/bin/setup.tt

```diff
@@ -15,6 +15,11 @@ FileUtils.chdir APP_ROOT do
   puts "== Installing dependencies =="
   system! "gem install bundler --conservative"
   system("bundle check") || system!("bundle install")
+<% if options.javascript != "importmap" -%>
```

Comment:
Can we copy the logic here: https://github.com/rails/rails/blob/ceb4b94baaf17f3a9f4ea795c83ec6c67211f737/railties/lib/rails/generators/app_base.rb#L392. We want to add this for these JS options only.

Reply:
thanks for the suggestion! 

I've updated the pr with it